pygments/lexers/shell.py

```diff
@@ -165,6 +168,7 @@ def get_tokens_unprocessed(self, text):
         for match in line_re.finditer(text):
             line = match.group()
             m = re.match(self._ps1rgx, line)
+            venv_re = re.match(self._venv, line)
```

Comment:
better use a precompiled regex. and the result should be e.g. `venv_match` not `venv_re`.

Reply:
Changed to a precompiled regex (should this also apply to `_ps1rgx`? That's a hefty one in comparison) and renamed as requested.

Reply:
yeah, ps1rgx should be precompiled too.

It's all a bit less important because re has a cache for regexes passed in as strings, but we're using lots of them and may exhaust the cache.